lib/zlib.js

```diff
@@ -238,8 +238,19 @@ const checkRangesOrGetDefault = hideStackFrames(
   }
 );
 
+const FLUSH_BOUND = [
+  [ Z_NO_FLUSH, Z_BLOCK ],
+  [ BROTLI_OPERATION_PROCESS, BROTLI_OPERATION_EMIT_METADATA ],
+];
+function isBrotliHandle(handle) {
+  return ((handle instanceof binding.BrotliEncoder) ||
+    (handle instanceof binding.BrotliDecoder)) ? 1 : 0;
+}
+
 // The base class for all Zlib-style streams.
 function ZlibBase(opts, mode, handle, { flush, finishFlush, fullFlush }) {
+  const isBrotli = isBrotliHandle(handle);
+
```

Comment:
You can just look at `mode`, no need to perform `instanceof` checks here

Reply:
done